apps/frontend/src/context/OptionsContext.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import useLocalStorage from '../hooks/useLocalStorage';
+
+type ReadWriteOptions = Options<{ value: boolean; set: (b: boolean) => void }>;
+
+type ReadOnlyOptions = Options<boolean>;
+
+type Options<T> = {
+  showIntervalTimer: T;
+  showTotalDurationTimer: T;
+};
+
+const OptionsContext = React.createContext<ReadWriteOptions | null>(null);
+
+const useOptionSetting = (key: string, initialValue: boolean) => {
+  const [value, setValue] = useLocalStorage<boolean>(key, initialValue);
+
+  return { value, set: setValue };
+};
+
+export const OptionsContextProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const showIntervalTimer = useOptionSetting('showIntervalTimer', true);
+
+  const showTotalDurationTimer = useOptionSetting(
+    'showTotalDurationTimer',
+    true
+  );
+
+  return (
+    <OptionsContext.Provider
+      value={{
+        showIntervalTimer,
+        showTotalDurationTimer,
+      }}
+    >
+      {children}
+    </OptionsContext.Provider>
+  );
+};
+
+export const useReadWriteOptions: () => ReadWriteOptions = () => {
+  const context = React.useContext(OptionsContext);
+  if (context === null) {
+    throw new Error(
+      'useReadWriteOptions must be used within a OptionsContextProvider'
+    );
+  }
+  return context;
+};
+
+export const useReadOnlyOptions: () => ReadOnlyOptions = () => {
+  const context = React.useContext(OptionsContext);
+  if (context === null) {
+    throw new Error(
+      'useReadOnlyOptions must be used within a OptionsContextProvider'
+    );
+  }
+  const result = {} as ReadOnlyOptions;
+
+  Object.entries(context).forEach(
+    ([k, v]) => (result[k as keyof ReadOnlyOptions] = v.value)
+  );
+
+  return result;
+};
```

Comment:
The idea is that basically only the OptionModal should have write-access

All the other components only needs read access. (and also makes it much cleaner to use, since it removes one layer)

Not the prettiest code, but.
Open for changes or improvements.
Could maybe be based on argument 'read' | 'readwrite' and some type based on that

Could also avoid make the code as simple as possible and just have {showIntervaltime:options.showIntervalTimer.value} for every field

wdyt?